Thanks again! Using master branch for liblinphone resulted in a chain reaction that I needed to use the master branch for four other linphone-related packages, but then I got past those errors I had. Yey!

Now I'm just wondering if it's ok to use unstable versions for all these packages (linphone, liblinphone, belr, belle-sip, mediastreamer and ortp) in nixpkgs. It seems to me like the easiest route now. So I guess that's ok. (I'll clean the fixup and squash commits before this is merged.)

Now the build process has some problems with Minizip:

```
Scanning dependencies of target linphone-qt
make[2]: *** No rule to make target 'minizip_OUTPUT/nix/store/v5njiq0qmjs1l6npv72cw2rbnxasrkd2-linphone-desktop-unstable-2020-03-06/lib/libminizip.so', needed by 'linphone'.  Stop.
```

That's looks really weird.. I don't even understand why it tries to make minizip because before these lines it has said that minizip found.

Well, if I remove this line from the cmake file, the build process doesn't stop here anymore: https://gitlab.linphone.org/BC/public/linphone-desktop/blob/master/CMakeLists.txt#L520


But then it fails at the very final steps when inking of Minizip to Linphone executable:
```
[100%] Linking CXX executable linphone
/nix/store/m6bb8cbdmb5yirxm8pa9ah9xri4557v7-binutils-2.31.1/bin/ld: CMakeFiles/app-library.dir/src/components/file/FileExtractor.cpp.o: in function `FileExtractor::~FileExtractor()':
/build/linphone-desktop-971997e162558d37051f89c9c34bbc240135f704/src/components/file/FileExtractor.cpp:41: undefined reference to `mz_stream_bzip_close'
/nix/store/m6bb8cbdmb5yirxm8pa9ah9xri4557v7-binutils-2.31.1/bin/ld: /build/linphone-desktop-971997e162558d37051f89c9c34bbc240135f704/src/components/file/FileExtractor.cpp:42: undefined reference to `mz_stream_bzip_delete'
/nix/store/m6bb8cbdmb5yirxm8pa9ah9xri4557v7-binutils-2.31.1/bin/ld: /build/linphone-desktop-971997e162558d37051f89c9c34bbc240135f704/src/components/file/FileExtractor.cpp:46: undefined reference to `mz_stream_posix_close'
/nix/store/m6bb8cbdmb5yirxm8pa9ah9xri4557v7-binutils-2.31.1/bin/ld: /build/linphone-desktop-971997e162558d37051f89c9c34bbc240135f704/src/components/file/FileExtractor.cpp:47: undefined reference to `mz_stream_posix_delete'
/nix/store/m6bb8cbdmb5yirxm8pa9ah9xri4557v7-binutils-2.31.1/bin/ld: CMakeFiles/app-library.dir/src/components/file/FileExtractor.cpp.o: in function `FileExtractor::clean()':
/build/linphone-desktop-971997e162558d37051f89c9c34bbc240135f704/src/components/file/FileExtractor.cpp:41: undefined reference to `mz_stream_bzip_close'
/nix/store/m6bb8cbdmb5yirxm8pa9ah9xri4557v7-binutils-2.31.1/bin/ld: /build/linphone-desktop-971997e162558d37051f89c9c34bbc240135f704/src/components/file/FileExtractor.cpp:42: undefined reference to `mz_stream_bzip_delete'
/nix/store/m6bb8cbdmb5yirxm8pa9ah9xri4557v7-binutils-2.31.1/bin/ld: /build/linphone-desktop-971997e162558d37051f89c9c34bbc240135f704/src/components/file/FileExtractor.cpp:46: undefined reference to `mz_stream_posix_close'
/nix/store/m6bb8cbdmb5yirxm8pa9ah9xri4557v7-binutils-2.31.1/bin/ld: /build/linphone-desktop-971997e162558d37051f89c9c34bbc240135f704/src/components/file/FileExtractor.cpp:47: undefined reference to `mz_stream_posix_delete'
/nix/store/m6bb8cbdmb5yirxm8pa9ah9xri4557v7-binutils-2.31.1/bin/ld: CMakeFiles/app-library.dir/src/components/file/FileExtractor.cpp.o: in function `FileExtractor::extract()':
/build/linphone-desktop-971997e162558d37051f89c9c34bbc240135f704/src/components/file/FileExtractor.cpp:41: undefined reference to `mz_stream_bzip_close'
/nix/store/m6bb8cbdmb5yirxm8pa9ah9xri4557v7-binutils-2.31.1/bin/ld: /build/linphone-desktop-971997e162558d37051f89c9c34bbc240135f704/src/components/file/FileExtractor.cpp:42: undefined reference to `mz_stream_bzip_delete'
/nix/store/m6bb8cbdmb5yirxm8pa9ah9xri4557v7-binutils-2.31.1/bin/ld: /build/linphone-desktop-971997e162558d37051f89c9c34bbc240135f704/src/components/file/FileExtractor.cpp:46: undefined reference to `mz_stream_posix_close'
/nix/store/m6bb8cbdmb5yirxm8pa9ah9xri4557v7-binutils-2.31.1/bin/ld: /build/linphone-desktop-971997e162558d37051f89c9c34bbc240135f704/src/components/file/FileExtractor.cpp:47: undefined reference to `mz_stream_posix_delete'
/nix/store/m6bb8cbdmb5yirxm8pa9ah9xri4557v7-binutils-2.31.1/bin/ld: /build/linphone-desktop-971997e162558d37051f89c9c34bbc240135f704/src/components/file/FileExtractor.cpp:60: undefined reference to `mz_stream_posix_create'
/nix/store/m6bb8cbdmb5yirxm8pa9ah9xri4557v7-binutils-2.31.1/bin/ld: /build/linphone-desktop-971997e162558d37051f89c9c34bbc240135f704/src/components/file/FileExtractor.cpp:65: undefined reference to `mz_stream_posix_open'
/nix/store/m6bb8cbdmb5yirxm8pa9ah9xri4557v7-binutils-2.31.1/bin/ld: /build/linphone-desktop-971997e162558d37051f89c9c34bbc240135f704/src/components/file/FileExtractor.cpp:69: undefined reference to `mz_stream_bzip_create'
/nix/store/m6bb8cbdmb5yirxm8pa9ah9xri4557v7-binutils-2.31.1/bin/ld: /build/linphone-desktop-971997e162558d37051f89c9c34bbc240135f704/src/components/file/FileExtractor.cpp:72: undefined reference to `mz_stream_bzip_open'
/nix/store/m6bb8cbdmb5yirxm8pa9ah9xri4557v7-binutils-2.31.1/bin/ld: /build/linphone-desktop-971997e162558d37051f89c9c34bbc240135f704/src/components/file/FileExtractor.cpp:76: undefined reference to `mz_stream_set_base'
/nix/store/m6bb8cbdmb5yirxm8pa9ah9xri4557v7-binutils-2.31.1/bin/ld: CMakeFiles/app-library.dir/src/components/file/FileExtractor.cpp.o: in function `FileExtractor::handleExtraction()':
/build/linphone-desktop-971997e162558d37051f89c9c34bbc240135f704/src/components/file/FileExtractor.cpp:256: undefined reference to `mz_stream_bzip_read'
/nix/store/m6bb8cbdmb5yirxm8pa9ah9xri4557v7-binutils-2.31.1/bin/ld: /build/linphone-desktop-971997e162558d37051f89c9c34bbc240135f704/src/components/file/FileExtractor.cpp:263: undefined reference to `mz_stream_bzip_get_prop_int64'
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/linphone-qt.dir/build.make:239: linphone] Error 1
make[1]: *** [CMakeFiles/Makefile2:174: CMakeFiles/linphone-qt.dir/all] Error 2
make: *** [Makefile:152: all] Error 2
builder for '/nix/store/bzc0afjh58p6miswyfzms0yk41b79id8-linphone-desktop-unstable-2020-03-06.drv' failed with exit code 2
```

I'm not quite sure why this happens because I have Minizip in `buildInputs`. I also checked with `nm -D` that this shared library contains (at least some of) these references the error message complains about. I tried giving CMake flags `-DMINIZIP_LIBRARIES=${minizip2}/lib/libminizip.so` or `-DMINIZIP_LIBRARIES=${minizip2}/lib` but it didn't help (not sure if it should be even needed to pass such a flag).

One option could be to let linphone build its own minizip. Hmm..

--- a/pkgs/development/libraries/ortp/default.nix
+++ b/pkgs/development/libraries/ortp/default.nix
@@ -1,15 +1,26 @@
-{ stdenv, cmake, fetchFromGitHub, bctoolbox }:
+{ bctoolbox
+, cmake
+, fetchFromGitLab
+, stdenv
+}:
 
 stdenv.mkDerivation rec {
   pname = "ortp";
-  version = "1.0.2";
+  # Using master branch for linphone-desktop caused a chain reaction that many
+  # of its dependencies needed to use master branch too.
+  version = "unstable-2020-03-17";
 
-  src = fetchFromGitHub {
-    owner = "BelledonneCommunications";
+  src = fetchFromGitLab {
+    domain = "gitlab.linphone.org";
+    owner = "public";
+    group = "BC";
     repo = pname;
-    rev = version;
-    sha256 = "12cwv593bsdnxs0zfcp07vwyk7ghlz2wv7vdbs1ksv293w3vj2rv";
+    rev = "804dfc4f90d1a4301127c7af10a74fd2935dd5d8";
+    sha256 = "1yr8j8am68spyy5d9vna8zcq3qn039mi16cv9jf5n4chs9rxf7xx";
   };
+
+  # Do not build static libraries
+  cmakeFlags = [ "-DENABLE_STATIC=NO" ];
 
   NIX_CFLAGS_COMPILE = "-Wno-error=stringop-truncation";
 
@@ -18,8 +29,9 @@ stdenv.mkDerivation rec {
 
   meta = with stdenv.lib; {
     description = "A Real-Time Transport Protocol (RFC3550) stack";
-    homepage = https://linphone.org/technical-corner/ortp;
-    license = licenses.gpl2Plus;
+    homepage = "https://linphone.org/technical-corner/ortp";
+    license = licenses.gpl3;
     platforms = platforms.all;
+    maintainers = with maintainers; [ jluttine ];
   };
 }

--- a/pkgs/development/libraries/soci/default.nix
+++ b/pkgs/development/libraries/soci/default.nix
@@ -1,0 +1,31 @@
+{ cmake
+, fetchFromGitHub
+, sqlite
+, stdenv
+}:
+
+stdenv.mkDerivation rec {
+  pname = "soci";
+  version = "4.0.0";
+
+  src = fetchFromGitHub {
+    owner = "SOCI";
+    repo = pname;
+    rev = version;
+    sha256 = "06faswdxd2frqr9xnx6bxc7zwarlzsbdi3bqpz7kwdxsjvq41rnb";
+  };
+
+  # Do not build static libraries
+  cmakeFlags = [ "-DSOCI_STATIC=OFF" ];
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ sqlite ];
+
+  meta = with stdenv.lib; {
+    description = "Database access library for C++";
+    homepage = "http://soci.sourceforge.net/";
+    license = licenses.boost;
+    platforms = platforms.all;
+    maintainers = with maintainers; [ jluttine ];
+  };
+}

--- a/pkgs/tools/misc/bcunit/default.nix
+++ b/pkgs/tools/misc/bcunit/default.nix
@@ -1,22 +1,30 @@
-{stdenv, fetchFromGitHub, cmake}:
+{ cmake
+, fetchFromGitLab
+, stdenv
+}:
+
 stdenv.mkDerivation rec {
-  name = "${baseName}-${version}";
-  baseName = "bcunit";
-  version = "3.0.2";
-  buildInputs = [cmake];
-  src = fetchFromGitHub {
-    owner = "BelledonneCommunications";
-    repo = baseName;
-    rev = version;
-    sha256 = "063yl7kxkix76r49qrj0h1qpz2p538d1yw8aih0x4i47g35k00y7";
+  pname = "bcunit";
+  # Latest release 3.0.2 is missing some functions needed by bctoolbox. See:
+  # https://gitlab.linphone.org/BC/public/bcunit/issues/1
+  version = "unstable-2019-11-19";
+
+  buildInputs = [ cmake ];
+  src = fetchFromGitLab {
+    domain = "gitlab.linphone.org";
+    owner = "public";
+    group = "BC";
+    repo = pname;
+    rev = "3c720fbf67dd3c02b0c7011ed4036982b2c93532";
+    sha256 = "1237hpmkls2igp60gdfkbknxpgwvxn1vmv2m41vyl25xw1d3g35w";
   };
 
-  meta = {
+  meta = with stdenv.lib; {
     inherit version;
-    description = ''A fork of CUnit test framework'';
-    license = stdenv.lib.licenses.lgpl2Plus;
-    maintainers = [stdenv.lib.maintainers.raskin];
-    platforms = stdenv.lib.platforms.linux;
+    description = "A fork of CUnit test framework";
+    homepage = "https://gitlab.linphone.org/BC/public/bcunit";
+    license = licenses.lgpl2Plus;
+    maintainers = with maintainers; [ raskin jluttine ];
+    platforms = platforms.linux;
   };
 }
-

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -22009,6 +22009,8 @@ in
 
   skanlite = libsForQt5.callPackage ../applications/office/skanlite { };
 
+  soci = callPackage ../development/libraries/soci { };
+
   sonic-lineup = libsForQt5.callPackage ../applications/audio/sonic-lineup { };
 
   sonic-visualiser = libsForQt5.callPackage ../applications/audio/sonic-visualiser { };

